vmdb/lib/miq_automation_engine/engine/miq_ae_state_machine.rb

```diff
@@ -0,0 +1,206 @@
+module MiqAeEngine
+  module MiqAeStateMachine
+    def process_state(f, message, args)
+      Benchmark.current_realtime[:state_count] += 1
```

Comment:
I see this was moved from the other code, but what is this used for?  It seems really bad to store the state count here, unless it's meant for calculations of some sort.